internal/app/go-filecoin/storage_market_connector/provider.go

```diff
@@ -81,24 +88,31 @@ func (s *StorageProviderNodeConnector) EnsureFunds(ctx context.Context, addr add
 
 // PublishDeals publishes storage deals on chain
 func (s *StorageProviderNodeConnector) PublishDeals(ctx context.Context, deal storagemarket.MinerDeal) (storagemarket.DealID, cid.Cid, error) {
-	sig := types.Signature(deal.Proposal.ProposerSignature.Data)
-
-	fcStorageProposal := types.StorageDealProposal{
-		PieceRef:  deal.Proposal.PieceRef,
-		PieceSize: types.Uint64(deal.Proposal.PieceSize),
-
-		Client:   deal.Proposal.Client,
-		Provider: deal.Proposal.Provider,
-
-		ProposalExpiration: types.Uint64(deal.Proposal.ProposalExpiration),
-		Duration:           types.Uint64(deal.Proposal.Duration),
-
-		StoragePricePerEpoch: types.Uint64(deal.Proposal.StoragePricePerEpoch.Uint64()),
-		StorageCollateral:    types.Uint64(deal.Proposal.StorageCollateral.Uint64()),
-
-		ProposerSignature: &sig,
+	fcStorageProposal := market.DealProposal{
+		PieceCID:             commcid.PieceCommitmentV1ToCID(deal.Proposal.PieceRef),
+		PieceSize:            aabi.PaddedPieceSize(deal.Proposal.PieceSize),
+		Client:               deal.Proposal.Client,
+		Provider:             deal.Proposal.Provider,
+		StartEpoch:           0,             // TODO populate when the miner module provides this value
+		EndEpoch:             math.MaxInt32, // TODO populate when the miner module provides this value
```

Comment:
@ingar will your work provide these values?

Reply:
Yes.  I have changed it over to use the `ClientDealProposal` from specs-actors, but I haven't changed it to use `PublishStorageDealsParams`.

Reply:
These values come from the `ClientDealProposal` embedded in the `deal` parameter, I believe.  So whatever the client proposes, it will be here.